src/components/locator.ts

```diff
@@ -296,6 +294,20 @@ export class Locator {
                     break
                 }
             }
+
+            const row = record.line - 1
+            let col = record.column < 0 ? 0 : record.column
+            // columns are typically not supplied by SyncTex, this could change in the future for some engines though
+            if (col === 0) {
+                const line = doc.lineAt(row)
+                if (data.textBeforeSelection !== '' && line.text.indexOf(data.textBeforeSelection) > -1) {
+                    col = line.text.indexOf(data.textBeforeSelection) + data.textBeforeSelection.length
+                } else if (data.textAfterSelection !== '' && line.text.indexOf(data.textAfterSelection) > -1) {
+                    col = line.text.indexOf(data.textAfterSelection)
+                }
```

Comment:
The failure example I mention in my post is mostly related to the use of `else if`.  If the `textBeforeSelection` is matched, then we do not use the `textAfterSelection`. Maybe, we could try to use both to further improve accuracy. I can make more tests if needed.